### PR TITLE
New event wait review

### PR DESCRIPTION
**Pull request by zhums**

### CODE REVIEW

File: src/core/inc/runtime.h [hunk 1]
```diff
@@ -361,7 +362,12 @@ class Runtime {
 
   uint64_t sys_clock_freq() const { return sys_clock_freq_; }
 
-  void KfdVersion(const HsaVersionInfo& version) { kfd_version.version = version; }
+  void KfdVersion(const HsaVersionInfo& version) {
+    kfd_version.version = version;
+    if (version.KernelInterfaceMajorVersion > 1 ||
```
Comment (fxkamd) [inline src/core/inc/runtime.h:367]: I think we should check for MajorVersion == 1 && MinorVersion >= 14. Any major version other than 1 is unsupported as a major version bump would indicate a break in backwards compatibility. I do not expect that we will ever bump the major version on the KFD API in the upstream driver. But if we do, it'll be for a very different ROCr.

File: src/core/runtime/interrupt_signal.cpp [hunk 2]
```diff
@@ -144,8 +144,14 @@ hsa_signal_value_t InterruptSignal::WaitRelaxed(
...
   uint32_t prior = waiting_++;
   MAKE_SCOPE_GUARD([&]() { waiting_--; });
-  // Allow only the first waiter to sleep (temporary, known to be bad).
-  if (prior != 0) wait_hint = HSA_WAIT_STATE_ACTIVE;
+
+  uint64_t event_age = 1;
+
+  if (!core::Runtime::runtime_singleton_->KfdVersion().supports_event_age) {
+      event_age = 0;
+      // Allow only the first waiter to sleep (temporary, known to be bad).
```
Comment (fxkamd) [inline src/core/runtime/interrupt_signal.cpp:152]: The "(temporary ...)" part of the comment can be removed, or replaced with an explanation why this is needed when event age is not supported by KFD.

File: src/core/runtime/signal.cpp [hunk 3]
```diff
@@ -197,8 +197,9 @@ uint32_t Signal::WaitAny(uint32_t signal_count, const hsa_signal_t* hsa_signals,
     for (uint32_t i = 0; i < signal_count; i++) signals[i]->waiting_--;
   });
 
-  // Allow only the first waiter to sleep (temporary, known to be bad).
-  if (prior != 0) wait_hint = HSA_WAIT_STATE_ACTIVE;
+  if (!core::Runtime::runtime_singleton_->KfdVersion().supports_event_age)
+      // Allow only the first waiter to sleep (temporary, known to be bad).
```
Comment (fxkamd) [inline src/core/runtime/signal.cpp:201]: Same as above.

File: src/core/inc/runtime.h (hunk 1 above)
Comment (zhums) [reply src/core/inc/runtime.h:367]: Sure. I will update it

**Pull request closed without merging**